sites/frontend/lib/parse-capi/index.js

```diff
@@ -19,17 +19,42 @@ type props = {
 
 const defaultArgs = { config: {}, contentFields: {} };
 
-export default ({ contentFields, config }: props = defaultArgs) => ({
-    // here we create our own object of CAPI content on the 'CAPI' key
-    CAPI: {
-        headline: headline(get(config, 'page.headline', '')),
-        standfirst: standfirst(get(contentFields, 'fields.standfirst', '')),
-        main: main(get(contentFields, 'fields.main', '')),
-        body: body(
-            get(contentFields, 'fields.blocks.body', [])
-                .map(block => block.bodyHtml)
-                .filter(Boolean)
-                .join(''),
-        ),
-    },
-});
+// TODO: we shouldn't do any cleaning or data transformation here, but
+// instead pass through the required data exactly. Dotcom Rendering
+// should be just that - a rendering layer only.
+//
+// We *should* however validate the data we receive at this point.
```

Comment:
Regarding cleaning data, would we want to move the responsibility for cleaning HTML outside of this repository and back upstream?

I'm thinking cleaning embedded elements in the HTML provided by CAPI as an example. Pushing this further upstream and back into `frontend` would mean that a developer who needs to do this would need to pull `frontend`, make the cleaning changes in their local `frontend` instance and then point their local `dotcom-rendering` server to their local `frontend` instance to see these changes. This doesn't seem like an ideal workflow compared to running `dotcom-rendering` alone and making the changes there?

Reply:
Hmm, good point.

I was more thinking here of the data interface. I don't think we should extract and recreate a differently 'shaped' json object here. Although cleaning does make sense.

I do think we should validate the shape here though (with some kind of library) and immediately error the request if it doesn't conform.